listener/upload.go

```diff
@@ -254,8 +255,10 @@ func ensureReposInDB(tx *gorm.DB, repos []string) (repoIDs []int, added int64, e
 		repoObjs[i] = models.Repo{Name: repo}
 	}
 
-	txOnConflict := tx.Set("gorm:insert_option", "ON CONFLICT DO NOTHING")
-	err = database.BulkInsert(txOnConflict, repoObjs)
+	txOnConflict := tx.Clauses(clause.OnConflict{
+		DoNothing: true,
+	})
+	err = txOnConflict.Create(repoObjs).Error
```

Comment:
Why we remove BulkInsert here? I see it's used on line 287.

Reply:
I think it's better to use build in bulk insert. On line 287 there is problem when inserting empty slices, we can either make special case for empty slices and use build in Bulk Insert, or use our custom.